added banker's algorithm

asf89 opened the pull request:
Eu não coloquei o algoritmo para comparar o tempo de cada thread. Eu precisei adicionar mais dois parâmetros de entrada na função runner() e na função bankerAlgorithm(). São eles:
int num_threads;
const int NUMBER_OF_RESOURCES;

asf89 committed:
added banker's algorithm

OtacilioN reviewed (commented):
Oi @Snakedark,

Percebi que os novos parâmetros do runner aparentemente não estão sendo passados para a função runner. Você testou esse código? Se sim, pode me explicar como os parâmetros estão sendo passados?

Caso de fato não estejam sendo, talvez uma solução possível seja incorporar tais parâmetros à estrutura de Banker

asf89 committed:
modified code and banker's algorithm

asf89 commented:
Modifiquei o código. coloquei as variáveis num_threads e number_of_resources dentro da estrutura Banker.

asf89 committed:
added safe state confirmation.

asf89 committed:
modified returns of banker's algorithm.

Commit:
Fix #3 'segmentation fault' logo após entrar na seção crítica

- Add malloc to boolean array
- Remove global variables from function param
- Also prettified the file

OtacilioN reviewed (changes requested):
Atualmente a função de banker aloca recurso para todas as threads independente da thread que está requisitando o recurso, quando o resultado esperado é alocar somente para a thread que pediu, definida pelo clienteId

asf89 committed:
Banker's algorithm working.

asf89 reviewed (commented):
código funcionando. Agora é detectar bugs.

Commit:
Fix scanf acquiring value of file

Commit:
Fix remmoval of live_values

asf89 committed:
changed return of banker's algorithm.

asf89 committed:
changed return of banker's algorithm.

OtacilioN reviewed (approved):
Aprovado, necessita fixar a questão do deadlock
